### ISSUE TITLE
Error "undefined symbol" with ugbc.zx and some of the examples.

### ISSUE THREAD

**Issue by kounch** [spotlessmind1975/ugbasic#564]
Trying to compile using `ugbc.zx` with some of the examples like `graphics_plot_03.bas` and others throw errors like

```
hjnskmesdjafmch.asm:827: error: undefined symbol: CLIPX1
  ^---- (CLIPX1)
```

Looking for a fix, i uncommented

```
XCURS:      DEFB    0
YCURS:      DEFB    0
CLIPX1:    DEFW 0
CLIPY1:    DEFW 0
CLIPX2:    DEFW 255
CLIPY2:    DEFW 191
```

in `src/hw/zx/vars.asm`. But this changed now the errors to something like

```
hjnskmesdjafmch.asm:774: error: duplicate definition: CLIPX1
  ^---- CLIPX1: DEFW 0
```

Then, i observed that 

```
deploy( vars ...
```

and 

```
deploy( zxvars ...
```

where in several places of `src/hw/zx.c` and in `src/targets/zx/point_at.c`

So, after changing all lines to be only

```
deploy( zxvars, src_hw_zx_vars_asm );
```

The error disappears, and now more than twenty of the examples that failed compiling for zx, give no error. I am not sure if this is the right way to solve, but at least i hope that it gives some clues to a proper fix.

**Comment by spotlessmind1975** [comment]
Hello @kounch and thank you very much for the bug report, and especially for the analysis.

The underlined error (the lack of the `CLIPX1` variable) does not actually depend on whether it is commented out in the sources. Indeed, it is right that it is because on the platforms based on the Z80 there is a careful management of the RAM memory, as most of the targets with this processor load the code in ROM. Therefore, the variables are not defined in the same code segment to avoid them being "read only". This is why entering it results in the "duplicate definition" error.

Clearly, this would not apply for the ZX Spectrum, which is why your fix may also give a valid result: however, the code handles differences more subtly than standard compilers, and the solution you propose would not work, for example, on the ColecoVision.

After careful verification, it can be seen that the `CLIPX1` variable has been removed from the peephole optimizer because it was (erroneously) considered not read by any instruction. The decision depends on the fact that the statement is badly written and therefore the pattern matching fails to notice that the variable is read and used. It was enough to correct the instruction for everything to work again, with all the optimizations.

Let me know if that works for you too.

Thank you again!

**Comment by spotlessmind1975** [comment]
Released on COLDFIX v1.13 [rev. 20230215.2241]
